### PR TITLE
fix: implement proper auth check for editing paired devices

### PR DESCRIPTION
**Pull request by LuccaBitfly**
See: BEDS-863

### CODE REVIEW

File: backend/pkg/api/data_access/notifications.go [hunk 1]
```diff
@@ -1639,47 +1640,47 @@ func (d *DataAccessService) UpdateNotificationSettingsNetworks(ctx context.Conte
...
-	// TODO: This can be deleted when the API layer has an improved check for the device id
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return err
-	}
-	if rowsAffected == 0 {
-		return fmt.Errorf("device with id %v to update notification settings not found", pairedDeviceId)
+		WHERE id = $3`,
+		name, IsNotificationsEnabled, pairedDeviceId)
+	if err == sql.ErrNoRows {
```
Comment (Eisei24) [inline backend/pkg/api/data_access/notifications.go:1667]: **issue:** `ExecContext` does not return a `sql.ErrNoRows` if no rows were affected by either UPDATE or DELETE.

File: backend/pkg/api/data_access/notifications.go [hunk 2]
```diff
@@ -1639,47 +1640,47 @@ func (d *DataAccessService) UpdateNotificationSettingsNetworks(ctx context.Conte
...
-	// TODO: This can be deleted when the API layer has an improved check for the device id
-	rowsAffected, err := result.RowsAffected()
-	if err != nil {
-		return err
+func (d *DataAccessService) DeleteNotificationSettingsPairedDevice(ctx context.Context, pairedDeviceId uint64) error {
+	_, err := d.userWriter.ExecContext(ctx, `
+		DELETE FROM users_devices
+		WHERE id = $1`,
+		pairedDeviceId)
+	if err == sql.ErrNoRows {
```
Comment (Eisei24) [inline backend/pkg/api/data_access/notifications.go:1678]: **issue:** `ExecContext` does not return a `sql.ErrNoRows` if no rows were affected by either UPDATE or DELETE.

**Review by Eisei24 — CHANGES_REQUESTED** [review]
This does fix the issue of the BEDS by not even entering `UpdateNotificationSettingsPairedDevice` or `DeleteNotificationSettingsPairedDevice` if the id does not exist but the methods themselves don't work like that.

File: backend/pkg/api/data_access/notifications.go (hunk 1 above)
Comment (LuccaBitfly) [reply backend/pkg/api/data_access/notifications.go]: fixed, ptal

File: backend/pkg/api/data_access/notifications.go (hunk 2 above)
Comment (LuccaBitfly) [reply backend/pkg/api/data_access/notifications.go]: fixed, ptal

**Review by Eisei24 — APPROVED** [review]
lgtm

**Pull request merged**